Sources/Workspace/Destination.swift

```diff
@@ -180,33 +181,35 @@ public struct Destination: Encodable, Equatable {
 }
 
 extension Destination {
-
     /// Load a Destination description from a JSON representation from disk.
     public init(fromFile path: AbsolutePath, fileSystem: FileSystem = localFileSystem) throws {
-        let json = try JSON(bytes: fileSystem.readFileContents(path))
-        try self.init(json: json)
-    }
-}
-
-extension Destination: JSONMappable {
-
-    /// The current schema version.
-    static let schemaVersion = 1
-
-    public init(json: JSON) throws {
-
+        let decoder = JSONDecoder.makeWithDefaults()
+        let version = try decoder.decode(path: path, fileSystem: fileSystem, as: VersionInfo.self)
         // Check schema version.
-        guard try json.get("version") == Destination.schemaVersion else {
+        guard version.version == 1 else {
             throw DestinationError.invalidSchemaVersion
         }
-
+        let destination = try decoder.decode(path: path, fileSystem: fileSystem, as: DestinationInfo.self)
         try self.init(
-            target: Triple(json.get("target")),
-            sdk: AbsolutePath(json.get("sdk")),
-            binDir: AbsolutePath(json.get("toolchain-bin-dir")),
-            extraCCFlags: json.get("extra-cc-flags"),
-            extraSwiftCFlags: json.get("extra-swiftc-flags"),
-            extraCPPFlags: json.get("extra-cpp-flags")
+            target: destination.target.map{ try Triple($0) },
+            sdk: destination.sdk,
+            binDir: destination.binDir,
+            extraCCFlags: destination.extraCCFlags,
+            extraSwiftCFlags: destination.extraSwiftCFlags,
+            extraCPPFlags: destination.extraCCFlags
         )
     }
 }
+
+fileprivate struct VersionInfo: Codable {
+    let version: Int
+}
+
+fileprivate struct DestinationInfo: Codable {
+    let target: String?
+    let sdk: AbsolutePath?
+    let binDir: AbsolutePath
```

Comment:
This small change from this much larger pull changes the format for destination JSON files, away from what's documented here and here